compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -220,60 +223,124 @@ impl Into<rustc_errors::DiagnosticId> for TypeAnnotationNeeded {
 impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     pub fn extract_type_name(
         &self,
-        ty: Ty<'tcx>,
+        arg: GenericArg<'tcx>,
         highlight: Option<ty::print::RegionHighlightMode>,
     ) -> (String, Option<Span>, Cow<'static, str>, Option<String>, Option<&'static str>) {
```

Comment:
I know this is a pre-existing issue, but could you extract this result value into a struct? It's not possible to guess what these components mean from the signature.